config.go

```diff
@@ -0,0 +1,86 @@
+package tilenol
+
+import (
+	"os"
+
+	"gopkg.in/go-yaml/yaml.v2"
+)
+
+// Config is a YAML server configuration object
+type Config struct {
+	// Cache configures the tile server cache
+	Cache *CacheConfig `yaml:"cache"`
+	// Layers configures the tile server layers
+	Layers []LayerConfig `yaml:"layers"`
+}
+
+// LoadConfig loads the configuration from disk, and decodes it into a Config object
+func LoadConfig(configFile *os.File) (*Config, error) {
+	dec := yaml.NewDecoder(configFile)
+	dec.SetStrict(true)
```

Comment:
It's not absolutely necessary, but I have strict YAML parsing set because there are some slightly awkward corners in YAML (e.g. number/string type coercion, etc.) that I'd rather just avoid ambiguity altogether.